app/code/core/Mage/Review/controllers/CustomerController.php

```diff
@@ -66,6 +91,12 @@ public function indexAction()
 
     public function viewAction()
     {
+        $review = $this->_loadReview((int) $this->getRequest()->getParam('id'));
+        if (!$review) {
+            $this->_forward('noroute');
```

Comment:
I suggest to redirect to reviews list instead of 404 error page.

Reply:
would't you expect an additional information message when redirected to the list view? In which cases this could ever happen to regular users?

Reply:
Yes, why not. This can happen to regular users when a review is deleted.

Reply:
And when you are redirected to 404 error page, you loose the customer menu, I think it's not good.